notifier/src/test/kotlin/modules/JiraNotifierTest.kt

```diff
@@ -19,257 +19,195 @@
 
 package org.ossreviewtoolkit.notifier.modules
 
+import com.atlassian.jira.rest.client.api.IssueRestClient
+import com.atlassian.jira.rest.client.api.JiraRestClient
+import com.atlassian.jira.rest.client.api.OptionalIterable
+import com.atlassian.jira.rest.client.api.ProjectRestClient
+import com.atlassian.jira.rest.client.api.SearchRestClient
+import com.atlassian.jira.rest.client.api.domain.BasicIssue
+import com.atlassian.jira.rest.client.api.domain.Comment
 import com.atlassian.jira.rest.client.api.domain.Issue
+import com.atlassian.jira.rest.client.api.domain.IssueType
+import com.atlassian.jira.rest.client.api.domain.Project
+import com.atlassian.jira.rest.client.api.domain.SearchResult
+import com.atlassian.jira.rest.client.api.domain.Transition
 
-import com.github.tomakehurst.wiremock.WireMockServer
-import com.github.tomakehurst.wiremock.client.WireMock.aResponse
-import com.github.tomakehurst.wiremock.client.WireMock.equalTo
-import com.github.tomakehurst.wiremock.client.WireMock.exactly
-import com.github.tomakehurst.wiremock.client.WireMock.get
-import com.github.tomakehurst.wiremock.client.WireMock.matching
-import com.github.tomakehurst.wiremock.client.WireMock.post
-import com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor
-import com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo
-import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
-import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import io.atlassian.util.concurrent.Promise
 
 import io.kotest.core.spec.style.WordSpec
 import io.kotest.matchers.result.shouldBeFailure
 import io.kotest.matchers.result.shouldBeSuccess
 import io.kotest.matchers.shouldBe
 import io.kotest.matchers.string.shouldContain
-import io.kotest.matchers.types.shouldBeTypeOf
 
-import java.io.File
-import java.net.URI
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
 
-import org.ossreviewtoolkit.model.config.JiraConfiguration
+import java.net.URI
 
 class JiraNotifierTest : WordSpec({
-    val server = WireMockServer(
-        WireMockConfiguration.options()
-            .dynamicPort()
-            .usingFilesUnderDirectory(TEST_FILES_ROOT)
-    )
-
-    beforeSpec {
-        server.start()
-    }
-
-    afterSpec {
-        server.stop()
-    }
-
-    beforeEach {
-        server.resetAll()
-    }
-
     "createIssue" should {
         "succeed for valid input" {
-            val projectKey = "TEST"
-            val notifier = JiraNotifier(
-                JiraConfiguration("http://localhost:${server.port()}", "testuser", "testpassword")
-            )
+            val project = createProject()
+            val issueClient = mockk<IssueRestClient>()
```

Comment:
Just a suggestion: There is some redundancy in the tests for creating and preparing the mocks for the different clients. In such cases, I find it useful to have a test helper class that manages the mock objects involved and offers some basic functionality tests can make use of. For instance, here a class like `JiraClientHelper` could be defined with properties for the REST client, the issue client, and the project client. It could already prepare the REST client mock to return the other mocks. It could further offer convenience functions to prepare the mocks for creating issues or querying projects.

Reply:
I will consider this when adding tests in an upcoming PR. But it's always a trade-off, if the test code becomes too complex it increases the risk that it introduces it's own bugs.

Reply:
> if the test code becomes too complex it increases the risk that it introduces it's own bugs.

Agreed, but I believe @oheger-bosch's main point was about *reducing* complexity by removing redundancies.

Reply:
At least partly done: https://github.com/oss-review-toolkit/ort/pull/5179